src/pages/Swap/index.tsx

```diff
@@ -570,22 +570,23 @@ export function Swap({
         showCancel={true}
       />
       <SwapHeader autoSlippage={autoSlippage} />
-      <ConfirmSwapModal
-        isOpen={showConfirm}
-        trade={trade}
-        originalTrade={tradeToConfirm}
-        onAcceptChanges={handleAcceptChanges}
-        attemptingTxn={attemptingTxn}
-        txHash={txHash}
-        recipient={recipient}
-        allowedSlippage={allowedSlippage}
-        onConfirm={handleSwap}
-        swapErrorMessage={swapErrorMessage}
-        onDismiss={handleConfirmDismiss}
-        swapQuoteReceivedDate={swapQuoteReceivedDate}
-        fiatValueInput={fiatValueTradeInput}
-        fiatValueOutput={fiatValueTradeOutput}
-      />
+      {trade && (
+        <ConfirmSwapModal
+          isOpen={showConfirm}
```

Comment:
it's weird that show state of this modal here is controlled both by `trade` and `showConfirm` - can this just be:

```
{trade && showConfirm && <ConfirmSwapModal ... />}
```

and make the modal internally always show if it's rendered?

Reply:
it would also save computation by not running the hooks and preparing React components if the modal is not rendered